runtime/src/utils/signal.rs

```diff
@@ -62,13 +62,15 @@ use std::{
 ///     let (tx, rx) = oneshot::channel();
 ///     context.with_label("waiter").spawn(|context| async move {
 ///         // Wait for signal or sleep
-///         select_loop! {
-///             sig = &mut signal => {
-///                 println!("Received signal: {}", sig.unwrap());
-///                 break;
-///             },
-///             _ = context.sleep(Duration::from_secs(1)) => {},
-///         };
+///         loop {
```

Comment:
Somewhat confusing we don't just use runtime here (as it incorporates signaler directly). I suppose this, however, is the best way to demonstrate?